src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultUsernamePasswordCredentials.java

```diff
@@ -0,0 +1,69 @@
+package org.jenkinsci.plugins.azurekeyvaultplugin;
+
+import com.azure.security.keyvault.secrets.models.KeyVaultSecret;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.impl.Messages;
+import com.microsoft.jenkins.keyvault.BaseSecretCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.Util;
+import hudson.util.Secret;
+
+
+public class AzureKeyVaultUsernamePasswordCredentials extends BaseSecretCredentials implements StandardUsernamePasswordCredentials {
+    final protected CredentialsScope scope;
+    final protected String username;
+
+
+    public AzureKeyVaultUsernamePasswordCredentials(
+            CredentialsScope scope,
+            String id,
+            String username,
+            String description,
+            String servicePrincipalId,
+            String secretIdentifier) {
+        super(scope, id, description, servicePrincipalId, secretIdentifier);
+        this.scope = scope;
+        this.username = Util.fixNull(username);
+    }
+
+    @NonNull
+    @Override
+    public Secret getPassword() {
+        final KeyVaultSecret secretBundle = getKeyVaultSecret();
+        return Secret.fromString(secretBundle.getValue());
+    }
+
+    @NonNull
+    @Override
+    public String getUsername() {
+        return this.username;
+    }
+
+    @Override
+    public CredentialsScope getScope() {
+        return this.scope;
+    }
+
+    @Extension(ordinal = 1)
+    public static class DescriptorImpl extends BaseStandardCredentialsDescriptor {
+
+        /**
+         * {@inheritDoc}
+         */
+        @NonNull
+        @Override
+        public String getDisplayName() {
+            return Messages.UsernamePasswordCredentialsImpl_DisplayName();
```

Comment:
> [ERROR] org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultUsernamePasswordCredentials$DescriptorImpl:58 com/cloudbees/plugins/credentials/impl/Messages must not be used

Reply:
Where do you get this error? I haven't seen it anywhere..

Reply:
it's in the CI logs, `mvn clean install` should be able to reproduce it

Reply:
Indeed does. I'll look at how to get rid of it

Reply:
Before I dig into this, do you know how this is supposed to be handled?

Reply:
either copy paste the code or suppress it https://github.com/jenkinsci/lib-access-modifier/blob/master/access-modifier-suppressions/src/main/java/org/kohsuke/accmod/restrictions/suppressions/SuppressRestrictedWarnings.java

Reply:
Great, copied the code, as I think suppressing warnings should be avoided.